### PR TITLE
week-3-assessment-completed'

### PR DESCRIPTION
**Pull request by Jk0ng**
# Submitting Week 3 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Career Readiness
- [x] Coding Challenges
- [x] Interview Questions
- [x] React Debugging

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

**Comment by Jk0ng** [comment]
If Christ is reading this; last time I asked about not having enough time for the challenges, I was talking about I couldn't finish the daily in-class challenges. So i wonder how if it is an expectation for us to finish those outside of class? Not the weekly assessment. Thanks!

File: career-readiness.md [hunk 1]
```diff
@@ -9,14 +9,15 @@ Creating a technical portfolio website is a big undertaking. Portfolios are almo
 Wire frames are used to outline the basic structure of your site. Wire frames should be vague enough to give you flexibility in your development process but structured enough to ensure all appropriate elements are present. 
 
 Submit the link to your wire frames:
+https://www.figma.com/file/4XL35UCK1FRqnUsc5lo6ZP/Untitled?type=design&node-id=0%3A1&mode=design&t=K8uBySxhjE9CHQOB-1
```
Comment (sjproctor) [inline career-readiness.md:12]: Good start!

File: code-challenges.test.js [hunk 2]
```diff
@@ -20,9 +20,50 @@ const fibonacciLength1 = 6
 const fibonacciLength2 = 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+// describe ("fibonacciLength", () => {
+//   it("takes in a number (greater than 2) and returns an array containing the Fibonacci sequence.", ()=>{
+//     expect(fibonacciLength(6)).toEqual([1, 1, 2, 3, 5, 8])
+//     expect(fibonacciLength(10)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+//   })
+// })
+
```
Comment (sjproctor) [inline code-challenges.test.js:29]: Remember to put the test variables inside the scope of the test!

File: code-challenges.test.js [hunk 3]
```diff
@@ -20,9 +20,50 @@ const fibonacciLength1 = 6
...
+// return newArray
+
+// }
+// console.log(fibonacciLength(5))
+// console.log(fibonacciLength(6))
+// console.log(fibonacciLength(7))
+// console.log(fibonacciLength(8))
+// console.log(fibonacciLength(9))
+// console.log(fibonacciLength(10))
+// console.log(fibonacciLength(15))
```
Comment (sjproctor) [inline code-challenges.test.js:64]: Remember that the purpose of your test is to run your function. If you want to test more edge cases you are welcome

File: code-challenges.test.js [hunk 4]
```diff
@@ -51,6 +92,27 @@ const studyMinutesWeek2 = {
...
+// input: object
+//output: an array of object's values
+//  create a function that takes in an object 
+//  use Object.value()to extracts the values from the object and return them in an array
+// then use .sort() method and set up the comparsion function a - b to arrrange the new array in an ascending order 
+
+const studyMinutes = (object) => {
+  return Object.values(object).sort((a , b)=> a - b)
+}
+
```
Comment (sjproctor) [inline code-challenges.test.js:115]: Very concise!

**Review by sjproctor — APPROVED** [review]
Great work. Your react debugging and interview questions look great. Your code logic looks good and puts you at advanced beginner. I want to challenge you to really use the tests. The next step for you is to see the value of why you are writing tests and how to use them more effectively.
